service/src/main.rs

```diff
@@ -514,12 +514,16 @@ fn start_worker<E, T, D, InitializationHandler, WorkerModeProvider>(
 		send_extrinsic(register_xt(), &node_api2, &tee_accountid_clone, is_development_mode)
 	};
 
-	let register_enclave_block_hash = send_register_xt().unwrap();
+	// Todo: Can't unwrap here because the extrinsic is for some reason not found in the block
+	// even if it was successful: https://github.com/scs/substrate-api-client/issues/624.
+	let register_enclave_block_hash = send_register_xt();
+	let api_register_enclave_xt_header =
+		integritee_rpc_api.get_header(register_enclave_block_hash).unwrap().unwrap();
```

Comment:
This one is a bit of a bummer, do you see any issues with my hotfix here @brenzi?

Reply:
OMG. ugly, but not problematic IMO